sa/sa.go

```diff
@@ -1375,11 +1378,38 @@ func (ssa *SQLStorageAuthority) NewAuthorizations2(ctx context.Context, req *sap
 		if err != nil {
 			return nil, err
 		}
-		err = ssa.dbMap.Insert(am)
+
+		// Each authz needs a (?,?...), in the VALUES block
+		qs.WriteString(fmt.Sprintf("(%s),", strings.TrimRight(strings.Repeat(" ?,", 11), ",")))
+		// Must follow the order of authzFields
+		vals = append(vals, am.ID)
+		vals = append(vals, am.IdentifierType)
+		vals = append(vals, am.IdentifierValue)
+		vals = append(vals, am.RegistrationID)
+		vals = append(vals, am.Status)
+		vals = append(vals, am.Expires)
+		vals = append(vals, am.Challenges)
+		vals = append(vals, am.Attempted)
+		// AttemptedAt is not included in the authzFields
+		vals = append(vals, am.Token)
+		vals = append(vals, am.ValidationError)
+		vals = append(vals, am.ValidationRecord)
+	}
+	// The VALUES block question-string has a trailing comma, remove it
+	qs_trimmed := strings.TrimRight(qs.String(), ",")
+	query := fmt.Sprintf("INSERT INTO authz2 (%s) VALUES %s RETURNING id;", authzFields, qs_trimmed)
+	rows, err := ssa.dbMap.Db.QueryContext(ctx, query, vals...)
+	if err != nil {
+		return nil, fmt.Errorf("%s %s", err, query)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var id_field int64
+		err = rows.Scan(&id_field)
```

Comment:
Nit, this variable name isn't idiomatic.
```suggestion
		var idField int64
		err = rows.Scan(&idField)
```

Reply:
Aha, thanks! Got the snake_case above too.